src/Components/test/testassets/TestContentPackage/DeclarativePersistStateComponent.razor

```diff
@@ -0,0 +1,21 @@
+﻿<p>Application state is <span id="@KeyName">@Value</span></p>
+<p>Render mode: <span id="render-mode-@KeyName">@_renderMode</span></p>
+
+@code {
+    [Parameter, EditorRequired]
+    public string InitialValue { get; set; } = "";
+
+    [Parameter, EditorRequired]
+    public string KeyName { get; set; } = "";
+
+    [SupplyParameterFromPersistentComponentState]
+    public string Value { get; set; }
```

Comment:
Can it be combined with `[Parameter]` and supplied from code (I expect it does). What is the precedence order?
Use-case
1. Set the property as component attribute from calling component
2. Override the value in this component
3. Serialize over the wire
4. What value gets restored on the other side?

Reply:
Nope, something is either a parameter or a cascading value. It's discouraged to modify parameters after you've received them https://learn.microsoft.com/en-us/aspnet/core/blazor/components/overwriting-parameters?view=aspnetcore-9.0

The pattern in this case is to have a separate parameter like `Value` and `InitialValue` where `Value` is the SupplyParameterFromPersistentComponentState one and `InitialValue` is the `Parameter` one. Then within OnInitialized the pattern is `Value ??= InitialValue` to either continue using the restored value or use the initial one.

Reply:
Does it throw if a property has both `[Parameter]` and `[SupplyParameterFrom...]` or `[CascadingParameter]`? I think it does but would be nice to confirm for sure.

Reply:
It doesn't throw

Reply:
> Does it throw if a property has both [Parameter] and [SupplyParameterFrom...] or [CascadingParameter]? I think it does but would be nice to confirm for sure.

It does throw, I can try again, but I'm pretty sure it throws for other providers and the logic is common.

Reply:
Ok, I was right, but it throws at the time the properties are being set 
https://github.com/dotnet/aspnetcore/blob/main/src/Components/Components/src/Reflection/ComponentProperties.cs#L64
https://github.com/dotnet/aspnetcore/blob/main/src/Components/Components/src/Reflection/ComponentProperties.cs#L71